services/containerregistry/registries.go

```diff
@@ -439,22 +429,24 @@ func SetTokenData(d *schema.ResourceData, tokenProps cr.TokenProperties) error {
 	return nil
 }
 
-//nolint:golint
-func SetCredentials(credentials cr.Credentials) map[string]interface{} {
+// SetCredentials sets the credentials for the registry token. username can be set on read, password will be set only on create
+func SetCredentials(credentials cr.Credentials) map[string]any {
 
-	credentialsEntry := map[string]interface{}{}
-
-	utils.SetPropWithNilCheck(credentialsEntry, "username", credentials.Username)
+	credentialsEntry := map[string]any{}
+	credentialsEntry["username"] = credentials.Username
+	if credentials.Password != "" {
+		credentialsEntry["password"] = credentials.Password
```

Comment:
@cristiGuranIonos why we have both

```
credentialsEntry["password"] = credentials.Password
```

and 

```
utils.SetPropWithNilCheck(credentialsEntry, "password", credentials.Password)
```

I think the first one should be enough since the `password` will never be a pointer.